Sources/Sentry/SentryInstallation.m

```diff
@@ -7,7 +7,7 @@ @implementation SentryInstallation
 
 static NSString *volatile installationString;
 
-+ (NSString *)id
++ (NSString *)idWithCacheDirectoryPath:(NSString *)cacheDirectoryPath
```

Comment:
`m`: We can now pass in different values for `cacheDirectoryPath`, but the `installationString` stays the same once initialized. So depending on which `cacheDirectoryPath` gets passed in first you get different `installationString`. To fix that, I think we need a dictionary of `cacheDirectoryPath` to `installationString`.